Update figures to be brand-compliant.

Avalei opened the pull request:
Update the fonts, colors, and layouts to be brand-compliant and readable.

Signed-off-by: Deidre Casey <deidre.casey@nordicsemi.no>

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-5146
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

greg-fer reviewed (commented):
Not sure about the following images, not covered by this PR:
- State flow charts of Asset Tracker v2 (quite colorful images, not sure if all these colors are justified in a state flow chart...)

Also notes about the following images:
- ot-arch_2x.svg - This image is taken directly from openthread.io: https://openthread.io/platforms -- I think we could keep it as-is or add "adapted from openthread.io" in the caption.
- pelion_client_states.svg - Again a note about states. The gradual color transition would suggest that SUSPENDED is the most important state? I'd prefer to have all Pelion states colored the same way, to match the legend in the lower left corner. With these edits, only the REGISTERED state qualifies as Pelion state. I would also suggest to check with @pdunaj and @MarekPieta about application and CAF related state images.

Avalei commented on doc/nrf/dm_code_base.rst:
```diff
@@ -72,7 +72,7 @@ A :ref:`manifest repository <zephyr:west-manifests>`, `sdk-nrf`_, contains a fil
 This is equivalent to topology T2 in the :ref:`west documentation <zephyr:west-basics>`.
 
-.. figure:: images/ncs-west-repos.png
+.. figure:: images/ncs-west-repos.svg
```
Asset Tracker V2 - Do you know which states we should highlight? I did not create this so I'm not sure what the original intent was behind the colors.

ot-arch_2x.svg - Sure, that's a good suggestion. I will revert the image and add the caption.

pelion_client_states.svg - My thinking in changing the colors was to show a progression/change in states to guide the user's eyes through the diagram since it seemed a bit flat to use just one tone of blue. I can see how using the dark blue emphasizes the Suspended state too much though. What about using the dark blue for the Registered state and the lighter tones for the other states? I can update the color legend too.

ncs-west-repos.svg - Sure, I will add in another block and description for "Other modules".

greg-fer replied:
For Asset Tracker, I think your contact is @peknis . 
For perlion client states, I'm fine with lighter tones, unless @pdunaj disagrees.
Also, @emob-nordic , can you have a look at the CAF state diagrams?

peknis replied:
To get the right info about the Asset Tracker images from the SME, that would be perhaps @coderbyheart.

coderbyheart replied:
All images in [asset_tracker_v2](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/asset_tracker_v2/README.html) were made by @simensrostad.

greg-fer reviewed (commented):
Also:
- matter_nrfconnect_overview_simplified_ncs.svg was already created with the figure guidelines in mind, so please keep it as it was, as it is also used on https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/nrfconnect_platform_overview.md . Moreover, the new image introduces a mistake, suggesting that all Matter layers are built with GN.
- This PR might be a good opportunity to fix font display on Linux machines, as several devs have reported over time that the current font is overflowing.

greg-fer reviewed (commented):
I went through all images 1 by 1, and have the following comments:

b0_flash_layout.svg 
b0_mcuboot_flash_layout.svg 
- this being flash layout, the blocks should touch each other; otherwise the separated blocks might suggest different memory components
-- this probably means using lines to separate flash sections

caf_buttons_states.svg 
- the original image features orange arrows as they are related to the CONFIG_CAF_BUTTONS_PM_EVENTS triangles

matter_nrfconnect_overview_simplified_ncs.svg
- Application should belong to NCS and be renamed User application, as per original image

pelion_client_states.svg 
- a bit unclear to me why some of the images have arrows changed to light blue, but this one has them changed to black?

thread_platform_design_multi.svg
- GitHub suggests the new image has quite a lot of blank padding at the bottom

zephyr_netstack_openthread-rx_sequence.svg
- hard to see on the comparison, but I have the impression that the two bottom rectangles next to points 3, 2, and 1 are barely visible

zigbee_platform_design_nRF5340_multi.svg
- Zephyr integration and the radio driver rectangles are now shorter, but MPSL is not there to justify this change -- is this an intended change?

Avalei commented:
> I went through all images 1 by 1, and have the following comments:
>
> b0_flash_layout.svg
> b0_mcuboot_flash_layout.svg
> 
>     * this being flash layout, the blocks should touch each other; otherwise the separated blocks might suggest different memory components
>       -- this probably means using lines to separate flash sections
Fixed.

> caf_buttons_states.svg
> 
>     * the original image features orange arrows as they are related to the CONFIG_CAF_BUTTONS_PM_EVENTS triangles
Fixed.

> matter_nrfconnect_overview_simplified_ncs.svg
> 
>     * Application should belong to NCS and be renamed User application, as per original image
Fixed. I was wondering though, in the original design there are two color blocks in the legend representing _Matter_ and _Matter (build with GN)_, however, only _Matter (build with GN)_ is used. Shouldn't we only need one color to represent the blocks "Matter Stack" and "Zephyr integration layer"?

> pelion_client_states.svg
> 
>     * a bit unclear to me why some of the images have arrows changed to light blue, but this one has them changed to black?
I went back in and updated all the images to use black arrows and text as the default. It has more contrast on the white background and is easier to read. My thinking is that we can use colored lines and text to highlight specific flows of data or communication - so there may still be some images with colored arrows intentionally.

> thread_platform_design_multi.svg
> 
>     * GitHub suggests the new image has quite a lot of blank padding at the bottom
Fixed.
> zephyr_netstack_openthread-rx_sequence.svg
> 
>     * hard to see on the comparison, but I have the impression that the two bottom rectangles next to points 3, 2, and 1 are barely visible
Fixed.

> zigbee_platform_design_nRF5340_multi.svg
> 
>     * Zephyr integration and the radio driver rectangles are now shorter, but MPSL is not there to justify this change -- is this an intended change?
I updated this to reflect the original again. I should have asked earlier, but is the size of those two boxes correct? I noticed on the other zigbee_platform images it is shortened, so I think that is why I made the change.

mirjaku commented:
I wen through the figures and here’s my comments. I don’t know the content enough to comment on the use of title case, but maybe it should be checked if it’s used accurately in these figures. We should use sentence case in diagrams, and I’m pretty sure some of the title cases could be changed to sentence case. I really like your use of color and space. :)
•	The ends of some of the lines are visible and look a bit disconnected. Maybe try overdrawing the line behind the blocks to make the connections look more seamless:
o	Thread_external_commissioning.svg
o	Thread_on-mesh_commissioning.svg
•	Some of the texts have white backgrounds that can be removed:
o	bt_mesh_light_ctrl_nodes.svg
o	bt_mesh_light_ctrl_reg.svg
o	bt_mesh_light_ctrl_states.svg
o	caf_ble_state_transitions.svg
o	caf_buttons_states.svg
o	caf_led_effect_structure.svg
o	caf_sensor_states.svg
o	esb_fig3_prx_ptx_trans_fail.svg
o	ncs-west-repos.svg
o	nfc_overview.svg
o	nrf_desktop_motion_sensing.svg
o	pelion_client_states.svg
o	zigbee_platform_design_nRF53.svg
o	zigbee_platform_design_nRF5340_multi.svg
•	caf_ble_state_transitions.svg:
o	There’s quite a lot of explanatory text in the figure that should be in a paragraph instead of the figure.
•	Some elements have white borderlines that make the edges look a bit fuzzy. Removing the lines would make the edges look sharper:
o	caf_buttons_states.svg
o	caf_sensor_states.svg
o	ml_app_architecture.svg (just the legend)
o	zigbee_platform_design_nRF53.svg
o	zigbee_signal_handler_03_first_start.svg
o	zigbee_signal_handler_04_reboot.svg
o	zigbee_signal_handler_05_formation.svg
o	zigbee_signal_handler_06_steering.svg
o	zigbee_signal_handler_07_idle.svg
o	zigbee_signal_handler_08_deep_sleep.svg
o	zigbee_signal_handler_09_leave.svg
•	esb_fig1_star_network.svg:
o	The texts are in bold. Should be changed to regular weight.
•	esb_fig2_ptx_trans_ok.svg:
o	A tiny thing but some of the dotted arrows could be snapped to the centers of the blocks.
•	esb_packet_format.svg:
o	There’s a white block on the right – maybe that could be removed.
•	Registered trademark symbol should be added after Bluetooth in “Bluetooth Low Energy”. Not sure if LE should be spelled out:
o	bt_mesh_and_ble.svg
o	ieee802154_nrf53_multiprot_design.svg
o	nfc_negotiated_connection_handover.svg
o	nfc_static_connection_handover.svg
o	nrf_desktop_arch.svg
o	nrf_desktop_arch_dmouse.svg
o	nrf_desktop_arch_dongle.svg
o	nrf_desktop_arch_gmouse.svg
o	nrf_desktop_arch_kbd.svg
o	thread_platform_design_multi.svg
o	thread_platform_design_nRF53_multi.svg
o	zigbee_platform_design_multi.svg
o	zigbee_platform_design_nRF5340_multi.svg
•	ieee802154_nrf53_singleprot_design.svg:
o	Tiny thing but maybe center-align the arrow from OpenAmp to IEEE 802.15.4.
•	matter_nrfconnect_overview_simplified_ncs.svg:
o	Is the block on the top-left a legend? Maybe change it to similar style as bt_mesh_light_ctrl_composition.svg? Also, I cannot see the “matter” color or pattern used in the diagram – I cannot see “Matter” in the original either, just the other Matter…
•	Might be the clearest to mark the legends the same way as in bt_mesh_light_ctrl_composition.svg with the text next to the color:
o	bt_mesh_and_ble.svg
o	nrf_desktop_arch.svg
o	nrf_desktop_arch_dmouse.svg
o	nrf_desktop_arch_dongle.svg
o	nrf_desktop_arch_gmouse.svg
o	nrf_desktop_arch_kbd.svg
o	pelion_client_states.svg
•	pelion_client_memory.svg:
o	The two white blocks have border lines that should be removed. If the white area is not visible enough, it should be replaced with a different color. The background of SPM (optional) makes the text a bit hard to read. Maybe use a solid color instead?
•	thread_data_flow_rx_uart.svg:
o	Is buffer capitalized in two different ways on purpose?
•	Is there text missing from the two grey blocks? If they’re empty on purpose, maybe add a legend explaining what they are. The two arrows on the right could be straightened by adjusting the widths of the blocks to make room for the texts (try making the ones on the left narrower):
o	thread_data_flow_tx_spi.svg
o	thread_data_flow_tx_uart.svg

Avalei commented:
Thanks for the review @mirjaku . I made those changes except for the following:

caf_ble_state_transitions.svg
- I agree that there is too much text in this design. I only copied the original design and made visual updates. Since this PR is already so large I added a separate task in Jira to re-design this image with less text, and move most of the words into the actual document.

thread_data_flow_tx_spi.svg
thread_data_flow_tx_uart.svg
- I will need to ask @greg-fer or another Thread team member if there is information to put in these empty sections. They were empty in the original design, but perhaps we can include some information for better clarity.

melwee01 reviewed (approved):
Bit beyond my wheelhouse, but the preview branch looks good to me at least. 👍

greg-fer commented:
> I will need to ask @greg-fer or another Thread team member if there is information to put in these empty sections. They were empty in the original design, but perhaps we can include some information for better clarity.

Answered in a comment to the JIRA task.

greg-fer commented:
@Avalei , apologies for late edit request. Would it be possible to add the following information to the image headers on the following figures?

- zephyr_netstack_openthread-rx_sequence.svg -- "Receiving IPv6 packet" moved to a centered header position on the image, "RX data flow" added along the arrow to the left, and a small "Packet received" added on top of the arrow
- zephyr_netstack_openthread-tx_sequence.svg -- "Transmitting IPv6 packet" moved to a centered header position on the image, "TX data flow" added along the arrow to the left, and a small "Packet transmitted" added under the bottom of the arrow (under its point)

This is related to KRKNWK-9822.

pdunaj reviewed (approved):
File I checked look ok. Two notes:
See file `doc/nrf/images/~$$caf_ble_state_transitions.~vsdx` - it looks like a temporary file from your editor and probably needs to be removed.
Also it would be great if such review was split into several PRs or at least commits. It is hard to fine files that should be checked.

Other then that all images look beautiful and are very neat :)

pdunaj commented:
@Avalei I know I repeat the info but it would be good to confirm it or fix. Can you check if `doc/nrf/images/~$$caf_ble_state_transitions.~vsdx` is intentional?

Avalei commented:
> 
> 
> @Avalei I know I repeat the info but it would be good to confirm it or fix. Can you check if `doc/nrf/images/~$$caf_ble_state_transitions.~vsdx` is intentional?

@pdunaj Thanks for the review, I have just removed the temporary file from the commit.

gmarull reviewed (approved):
LGTM
